### DISCUSSION TITLE
Nextjs and seperate backend Nodejs server with Express.js

### DISCUSSION THREAD

**Discussion by Kepron** [nextauthjs/next-auth#11579] [Help]
Hi,

I'm editing my Nextjs project to migrate to the app folder. While doing this, I thought it would be good to get rid of redux, which was causing me a lot of problems, and decided to use auth.js. Due to build problems caused by SSG, I am using a separate nodejs project for the backend.

I don't want to use any API routes in Nextjs because I have an obsession with everything being neat and clean. I want to handle the API and server-side operations in the Nodejs project as much as possible.

**TL;DR, I want to do auth operations with Express.js on my Node.js server. But the rest of my project will be on Nextjs.**

- example.com - Port 3000 Nextjs server.
- api.example.com - Port 3001 Nodejs server

Is it possible to do this? If possible, can you share a short example that explains the logic?

**Comment by hillac** [comment]
I think this issue is related and has a similar setup to what you want: https://github.com/nextauthjs/next-auth/issues/10389#issuecomment-2172852578 . This [comment](https://github.com/nextauthjs/next-auth/issues/10389#issuecomment-2172852578) seems to suggest it can work as long as you ensure you install v5.

**Reply by Kepron** [reply]
I've seen them, but I don't think they're what I'm looking for. I'll log in with credentials. You'll need companycode, username, and password to log in.

**Reply by hillac** [reply]
So add a credentials provider?

**Reply by Kepron** [reply]
The repo in the link is quite complicated. Also, next-auth v4 is used. I want to use v5. It would be quite a problem to switch from v4 to v5 later.

**Reply by hillac** [reply]
It looks very straight forward. v5 next-auth front end, @auth/express backend, and configure the api rewrite in next config. I'm not suggesting you fork their repo.

**Reply by Kepron** [reply]
> It looks very straight forward. v5 next-auth front end, @auth/express backend, and configure the api rewrite in next config. I'm not suggesting you fork their repo.

Is it really that simple? But how will this work?

```
import { auth } from "../auth"
 
export default async function UserAvatar() {
  const session = await auth()
 
  if (!session.user) return null
 
  return (
    <div>
      <img src={session.user.image} alt="User Avatar" />
    </div>
  )
}
```

Since auth will be in nodejs, it will not be possible to use it this way.

**Reply by hillac** [reply]
I haven't tried so not sure. I'm just going off the comment of the person in that thread that said they were successful. Have a go and see. To get the session it will call the api which I'm guessing will get re-written to your node server.
